code/modules/mob/living/carbon/human/species_types/jellypeople.dm

```diff
@@ -479,88 +479,169 @@
+	if(istype(mind.current.get_item_by_slot(ITEM_SLOT_HEAD), /obj/item/clothing/head/foilhat))
 		if(owner)
-			to_chat(M, "<span class='danger'>[owner.real_name]'s no-good syndicate mind-slime is blocked by your protective headgear!</span>")
-
+			to_chat(mind.current, "<span class='danger'>[owner.real_name]'s no-good syndicate mind-slime is blocked by your protective headgear!</span>")
 		return FALSE
-	if(M in linked_mobs)
+	if(mind in linked_minds)
 		return FALSE
 	if(!owner)
 		return FALSE
-	linked_mobs.Add(M)
-	to_chat(M, "<span class='notice'>You are now connected to [owner.real_name]'s Slime Link.</span>")
+	linked_minds.Add(mind)
 	var/datum/action/innate/linked_speech/action = new(src)
 	linked_actions.Add(action)
-	action.Grant(M)
+	action.Grant(mind.current)
+	to_chat(mind.current, "<span class='notice'>You are now connected to [owner.real_name]'s Slime Link.</span>")
+	GLOB.slime_linked_with[mind] = WEAKREF(src)
+	RegisterSignal(mind.current, COMSIG_MOB_LOGIN, PROC_REF(login_notify))
+	RegisterSignal(mind.current, COMSIG_MOB_DEATH, PROC_REF(link_death))
+	RegisterSignal(mind, COMSIG_MIND_TRANSFER_TO, PROC_REF(mind_transfer))
+	to_chat(mind.current, "<span class='big notice'>You can use :[MODE_KEY_SLIMELINK] or .[MODE_KEY_SLIMELINK] to talk over the slime link!</span>")
 	return TRUE
 
-/datum/species/oozeling/stargazer/proc/unlink_mob(mob/living/M)
-	var/link_id = linked_mobs.Find(M)
-	if(!(link_id))
+/datum/species/oozeling/stargazer/proc/unlink_mind(datum/mind/mind)
+	var/link_id = linked_minds.Find(mind)
+	if(!link_id)
 		return
-	var/datum/action/innate/linked_speech/action = linked_actions[link_id]
-	action.Remove(M)
 	var/mob/living/carbon/human/owner = slimelink_owner.resolve()
-	if(owner)
-		to_chat(M, "<span class='notice'>You are no longer connected to [owner.real_name]'s Slime Link.</span>")
-	linked_mobs -= M
+	var/datum/action/innate/linked_speech/action = linked_actions[link_id]
+	if(mind.current)
+		action.Remove(mind.current)
+		to_chat(mind.current, "<span class='notice'>You are no longer connected to [owner.real_name]'s Slime Link.</span>")
+		UnregisterSignal(mind.current, list(COMSIG_MOB_LOGIN, COMSIG_MOB_DEATH))
+	UnregisterSignal(mind, COMSIG_MIND_TRANSFER_TO)
 	linked_actions -= action
-	qdel(action)
+	linked_minds -= mind
+	if(GLOB.slime_linked_with[mind])
+		var/datum/weakref/mind_ref = GLOB.slime_linked_with[mind]
+		if(mind_ref?.resolve() == src)
+			GLOB.slime_linked_with -= mind
+
+/datum/species/oozeling/stargazer/proc/slime_chat(mob/living/user, message)
+	var/mob/living/carbon/human/link_owner = slimelink_owner.resolve()
+	if(!link_owner || !istype(link_owner))
+		return
+	if(!user || !user.mind)
+		return
+	if(!user == link_owner && !(user.mind in linked_minds))
+		to_chat(user, "<span class='warning'>The link seems to have been severed...</span>")
+		return
+	if(!length(message))
+		return
+	message = user.treat_message_min(message)
```

Comment:
doesn't this mean chats from tgui say are unfiltered?

Reply:
No? Everything that calls `slime_chat` has a `CHAT_FILTER_CHECK` some time before it.